axelor-stock/src/main/resources/views/StockMoveLine.xml

```diff
@@ -86,7 +88,13 @@
     <form name="stock-move-line-form" title="Stock move line" model="com.axelor.apps.stock.db.StockMoveLine"
     	onNew="action-stock-move-line-record-new" onLoad="action-stock-move-line-group-onload" width="large"
     	canNew="false" canDelete="false">
-        <panel name="main" >
+        
+   		<panel name="titleLinePanel">
+    		<field name="isTitleLine" onChange="action-stock-move-line-attrs-isTitleLine-onchange"/>
+    		<field name="productName" showIf="isTitleLine"/>
+    	</panel>
```

Comment:
Panel and boolean field should follow same design as on invoiceLine and purchaseOrder line .